ui/pages/SingleAsset.tsx

```diff
@@ -98,14 +98,23 @@ export default function SingleAsset(): ReactElement {
       localizedDecimalAmount: undefined,
     }
 
-  const isUntrusted = isUntrustedAsset(asset)
-  const isUnverifiedByUser = isUnverifiedAssetByUser(asset)
+  const isTrusted = isTrustedAsset(asset)
+  const isVerified = isVerifiedAsset(asset)
   const [warnedAsset, setWarnedAsset] =
     useState<SmartContractFungibleAsset | null>(null)
 
   const showActionButtons = isEnabled(FeatureFlags.SUPPORT_UNVERIFIED_ASSET)
-    ? !isUnverifiedByUser
+    ? isTrusted
     : true
+  const showVerificationButton = isEnabled(
+    FeatureFlags.SUPPORT_UNVERIFIED_ASSET
+  )
+    ? !isVerified && !isTrusted
+    : false
```

Comment:
If an asset is not trusted, then it must be a custom asset that has not been verified, hence we should be able to refactor this to:

```suggestion
const showVerificationButton =
    isEnabled(FeatureFlags.SUPPORT_UNVERIFIED_ASSET) && !isTrusted
```

Reply:
I actually think we should skip our easy booleans and tunnel in here. We should show the verification button only if `verified` is `undefined`: `asset?.metadata?.verified === undefined`.

My reasoning here is that, on the off chance that the user somehow finds themselves (through a bug!) viewing the single asset page for an asset that they have explicitly marked as not verified, we don't want to make it look exactly like a page whose trust level we don't actually know.

In a perfect world we would have some sort of “⚠️ Asset is not trusted” thing in the UI, but that's (1) out of scope here and (2),again, if they're viewing this asset page it means we've had a bug since we remove explicitly unverified assets from all lists that might lead you to this page.